runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -243,9 +363,277 @@ TR::ARM64PrivateLinkage::setParameterLinkageRegisterIndex(TR::ResolvedMethodSymb
+   cursor = generateMemSrc1Instruction(cg(), TR::InstOpCode::sturx, NULL, returnAddressMR, machine->getRealRegister(TR::RealRegister::lr), cursor);
+
+   // --------------------------------------------------------------------------
+   // Speculatively adjust Java SP with the needed frame size.
+   // This includes the preserved RA slot.
+   //
+   if (constantIsUnsignedImm12(alignedFrameSizeIncludingReturnAddress))
+      {
+      cursor = generateTrg1Src1ImmInstruction(cg(), TR::InstOpCode::subimmx, NULL, javaSP, javaSP, alignedFrameSizeIncludingReturnAddress, cursor);
+      }
+   else
+      {
+      TR::RealRegister *x9Reg = machine->getRealRegister(TR::RealRegister::RegNum::x9);
+
+      if (constantIsUnsignedImm16(alignedFrameSizeIncludingReturnAddress))
+         {
+         // x9 will contain the aligned frame size
+         //
+         cursor = loadConstant32(cg(), NULL, alignedFrameSizeIncludingReturnAddress, x9Reg, cursor);
+         cursor = generateTrg1Src2Instruction(cg(), TR::InstOpCode::subx, NULL, javaSP, javaSP, x9Reg, cursor);
+         }
+      else
+         {
+         TR_ASSERT_FATAL(0, "Large frame size not supported in prologue yet");
+         }
+      }
+
+   // --------------------------------------------------------------------------
+   // Perform javaSP overflow check
+   //
+   if (!comp()->isDLT())
+      {
+      //    if (javaSP < vmThread->SOM)
+      //       goto stackOverflowSnippetLabel
+      //
+      // stackOverflowRestartLabel:
+      //
+      TR::MemoryReference *somMR = new (cg()->trHeapMemory()) TR::MemoryReference(vmThread, cg()->getStackLimitOffset(), cg());
+      TR::RealRegister *somReg = machine->getRealRegister(TR::RealRegister::RegNum::x10);
+      cursor = generateTrg1MemInstruction(cg(), TR::InstOpCode::ldrimmx, NULL, somReg, somMR, cursor);
+
+      TR::RealRegister *zeroReg = machine->getRealRegister(TR::RealRegister::xzr);
+      cursor = generateTrg1Src2Instruction(cg(), TR::InstOpCode::subsx, NULL, zeroReg, javaSP, somReg, cursor);
+
+      TR::LabelSymbol *stackOverflowSnippetLabel = generateLabelSymbol(cg());
+      cursor = generateConditionalBranchInstruction(cg(), TR::InstOpCode::b_cond, NULL, stackOverflowSnippetLabel, TR::CC_LS, cursor);
+
+      TR::LabelSymbol *stackOverflowRestartLabel = generateLabelSymbol(cg());
+      cursor = generateLabelInstruction(cg(), TR::InstOpCode::label, NULL, stackOverflowRestartLabel, cursor);
+
+      cg()->addSnippet(new (cg()->trHeapMemory()) TR::ARM64StackCheckFailureSnippet(cg(), NULL, stackOverflowRestartLabel, stackOverflowSnippetLabel));
+      }
+
+   // --------------------------------------------------------------------------
+   // Preserve GPRs
+   //
+   // javaSP has been adjusted, so preservedRegs start at offset outgoingArgSize
+   // relative to the javaSP
+   //
+   // Registers are preserved in order from x21 (high memory) -> x28 (low memory)
```

Comment:
It is `from x21 (low memory) -> x28 (high memory)` now.

Reply:
Yes, thanks.